### ISSUE TITLE
ssidToNetworkId - check test.SSID before test.SSID.equals

### ISSUE THREAD

**Issue by tripflex** [arsenal942/Cordova-Network-Manager#3]
After pulling out my hair for hours, the error I kept getting was:
`Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference`

Which I tracked down to this bit of code:

```java
        List<WifiConfiguration> currentNetworks = wifiManager.getConfiguredNetworks();
        int networkId = -1;

        // For each network in the list, compare the SSID with the given one
        for (WifiConfiguration test : currentNetworks) {
            if ( test.SSID.equals(ssid) ) {
                networkId = test.networkId;
            }
        }
```

Which I ended up figuring out the issue was related to some null WiFi object (or one missing data) that was added by the Cerberus Android app (https://www.cerberusapp.com/), and only after I removed that entry did I no longer receive this error.

Maybe a good idea to just check if `test.SSID` is available and not null before checking `.equals`?  I honestly don't know how that got added, but just in case of instances where some erroneous wifi config exists, would be good to check first to prevent these errors.

Thanks for the great plugin though!

**Comment by arsenal942** [comment]
Hi @tripflex,
I will admit, for the Android platform, I have done next to no modifications from the original WifiWizard so that's probably why I missed it. I've been pretty busy with my IRL job however I'll get around to fixing this issue. 

Or you can always do a pull request and I'll merge everything. :D

**Comment by arsenal942** [comment]
Well I do take that back - I have modified how connect and disconnect works since I required it for my application of the plugin. Completely optional to give it the 'ssidToDisable' or 'ssidToEnable' parameter. This can probably be done better but some Android OS have an issue where they just never connect to the network you say to connect to and this was the only way I could get around it.

**Comment by tripflex** [comment]
@arsenal942 thanks for the quick reply, yeah it's really just some sanity checking before calling `.equals` ... but my Java knowledge is slim to none, otherwise I would have submitted a PR for it 👅 

Maybe something like `if( test.SSID && test.SSID !== null && test.SSID.equals(ssid)` ?

And yeah haven't got to the part of connecting to the network yet, should be at that point in an hour or two to test and see how it works out, but that was one thing i was wondering about .... should I just get the currently connected WiFi SSID and pass that as the one to disconnect from (when calling method to connect)?

**Comment by arsenal942** [comment]
@tripflex I've already done the fix so I'll issue a commit later today and publish to NPM accordingly. I've fixed a few other little bugs as well so I just need to do a bit of testing.

I would advise NOT using those parameters unless you observe the issue I am describing. If in the event you do, then yes. Disable the currently connected network, store it in some form of persistent memory (local storage) and then you can always re-enable that network once you are completed with everything.

If you would like a code sample, I can provide you with one. My discord is also on the read-me at the bottom. Nomm#0176

**Comment by arsenal942** [comment]
@tripflex Fixed in v2.1.1 - I confirmed it my end but if you can also confirm that it solves your problem, that'd be great. Once you have, I will update the version to 2.2.0 and publish as required.

**Comment by tripflex** [comment]
@arsenal942 yup i just updated my version with the null check and don't seem to have any issues 

Regarding the `androidConnectToNetwork` though, it looks like an SSID to disconnect from MUST be passed otherwise it will error out, so from what I can tell, that's not an "optional" argument.

Maybe it would be easier to just get the current SSID using the method you already have, no?

```java
    private boolean androidConnectNetwork(CallbackContext callbackContext, JSONArray data) {
        Log.d(TAG, "cordovaNetworkManager: connectNetwork entered.");
        if(!validateData(data)) {
            callbackContext.error("cordovaNetworkManager: connectNetwork invalid data");
            Log.d(TAG, "cordovaNetworkManager: connectNetwork invalid data.");
            return false;
        }
        String ssidToConnect = "";
		String currentSSID = "";

        try {
            ssidToConnect = data.getString(0);
			currentSSID = data.getString(1);
        }
        catch (Exception e) {
            callbackContext.error(e.getMessage());
            Log.d(TAG, e.getMessage());
            return false;
        }

        int networkIdToConnect = ssidToNetworkId(ssidToConnect);
		int currentNetworkId = ssidToNetworkId(currentSSID);

        if (networkIdToConnect >= 0 && currentNetworkId >= 0) {
            // We disable the network before connecting, because if this was the last connection before
            // a disconnect(), this will not reconnect.

			wifiManager.disableNetwork(currentNetworkId);
			wifiManager.enableNetwork(networkIdToConnect, true);

            SupplicantState supState;
            WifiInfo wifiInfo = wifiManager.getConnectionInfo();
            supState = wifiInfo.getSupplicantState();

            callbackContext.success(supState.toString());
            return true;

        }else{
            callbackContext.error("cordovaNetworkManager: Cannot connect to network");
            return false;
        }
    }
```

**Comment by tripflex** [comment]
I'll actually close this and open in new ticket specific to this
